File: backend/src/main/java/org/example/backend/model/Recipe.java
package org.example.backend.model;

import lombok.*;

import java.util.List;

@With
@Builder
@Data
public class Recipe {
    String id;
    String name;
    String category;
    String image;
    List<Ingredient> ingredients;
    String description;
    PreparationSpeed speed;
    String notes;
    String opinionOfTheDish;
    String linkToSource;
    Boolean favorite;
}

File: frontend/src/models/Recipe.ts
import type {DishCategory} from "./DishCategory.ts";
import type {Ingredient} from "./Ingredient.ts";
import type {PreparationSpeed} from "./PreparationSpeed.ts";

export type Recipe = {
    id: string,
    name: string,
    category: DishCategory,
    image: string,
    ingredients: Ingredient[],
    description: string,
    speed: PreparationSpeed,
    notes: string,
    opinionOfTheDish: string,
    linkToSource: string,
    favorite: boolean,
}